New `wgpu_examples::timestamp_queries::tests::timestamps_passes` test fails in CI

<!-- Thank you for filing this! Please read the [debugging tips](https://github.com/gfx-rs/wgpu/wiki/Debugging-wgpu-Applications).
That may let you investigate on your own, or provide additional information that helps us to assist.-->

**Description**
`wgpu_examples::timestamp_queries::tests::timestamps_passes` failed in CI environment.

**Repro steps**
See https://github.com/gfx-rs/wgpu/actions/runs/27740426584/job/82066194727?pr=9678

**Expected vs observed behavior**
Expected to pass, because PR https://github.com/gfx-rs/wgpu/pull/9678 doesn't change the behavior of non-unwinding code path.

**Extra materials**

```
     Summary [ 122.964s] 1326 tests run: 1325 passed, 1 failed, 12 skipped
        FAIL [   0.493s] ( 455/1326) wgpu-examples [Executed] [Gl/llvmpipe (LLVM 18.1.3, 256 bits)/1] wgpu_examples::timestamp_queries::tests::timestamps_passes
error: test run failed
error: process didn't exit successfully: `/home/runner/.rustup/toolchains/1.93-x86_64-unknown-linux-gnu/bin/cargo nextest run --manifest-path /home/runner/work/wgpu/wgpu/Cargo.toml --target-dir /home/runner/work/wgpu/wgpu/target/llvm-cov-target --benches --tests --all-features` (exit status: 100)
Error: Tests failed

Caused by:
    command exited with non-zero code `cargo llvm-cov --no-cfg-coverage --no-report nextest --benches --tests --all-features`: 100
Error: Process completed with exit code 1.
```

**Platform**
CI environment.

--- a/examples/features/src/timestamp_queries/mod.rs
+++ b/examples/features/src/timestamp_queries/mod.rs
@@ -430,7 +430,8 @@ pub fn main() {
 
 #[cfg(test)]
 pub mod tests {
-    use wgpu_test::{gpu_test, GpuTestConfiguration};
+    use wgpu::Backends;
+    use wgpu_test::{gpu_test, FailureCase, GpuTestConfiguration};
 
     use super::{submit_render_and_compute_pass_with_queries, QueryResults};
 
@@ -464,6 +465,11 @@ pub mod tests {
                     wgpu::Features::TIMESTAMP_QUERY
                         | wgpu::Features::TIMESTAMP_QUERY_INSIDE_ENCODERS
                         | wgpu::Features::TIMESTAMP_QUERY_INSIDE_PASSES,
+                )
+                .expect_fail(
+                    FailureCase::backend_adapter(Backends::GL, "llvmpipe")
+                        .panic("unexpected: inner timestamp before compute pass start timestamp")
+                        .flaky(),
                 ),
         )
         .run_sync(|ctx| test_timestamps(ctx, true, true));
@@ -504,7 +510,10 @@ pub mod tests {
             assert!(render_inside_timestamp <= render_start_end_timestamps[1]);
         }
         if let Some(compute_inside_timestamp) = compute_inside_timestamp {
-            assert!(compute_inside_timestamp >= compute_start_end_timestamps[0]);
+            assert!(
+                compute_inside_timestamp >= compute_start_end_timestamps[0],
+                "unexpected: inner timestamp before compute pass start timestamp"
+            );
             assert!(compute_inside_timestamp <= compute_start_end_timestamps[1]);
         }
     }

--- a/examples/features/src/timestamp_queries/mod.rs
+++ b/examples/features/src/timestamp_queries/mod.rs
@@ -475,13 +475,17 @@ pub mod tests {
     ) {
         let queries = submit_render_and_compute_pass_with_queries(&ctx.device, &ctx.queue);
         let raw_results = queries.wait_for_results(&ctx.device);
+        println!("Raw timestamp buffer contents: {raw_results:?}");
+        let query_results = QueryResults::from_raw_results(raw_results, timestamps_inside_passes);
+        query_results.print(&ctx.queue);
+
         let QueryResults {
             encoder_timestamps,
             render_start_end_timestamps,
             render_inside_timestamp,
             compute_start_end_timestamps,
             compute_inside_timestamp,
-        } = QueryResults::from_raw_results(raw_results, timestamps_inside_passes);
+        } = query_results;
 
         // Timestamps may wrap around, so can't really only reason about deltas!
         // Making things worse, deltas are allowed to be zero.

